### PR TITLE
TOML test vectors and unit tests for `get_pubkey`, `register_wallet`, `sign_message`

### PR DESCRIPTION
**Pull request by bigspider**
Extracts from the TOML test vectors for all the device operations from the python test suites.

Implements C tests for `get_extended_pubkey`, `register_wallet` and `sign_message`.
Since tests for address generation (at least most of it) was added in #493, `sign_psbt` is the only command not covered by C tests (apart from `get_master_fingerprint` that is more ore less trivial).

Some edge cases that were not covered by python tests are now tested.

### CODE REVIEW

**Review by 3697503 — APPROVED** [review]

File: test_vectors/README.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+All vectors were generated against the Speculos test seed (see
+[`fixtures.toml`](fixtures.toml)) on the **testnet** network.
+
+## Files
+
+| File | Operation |
+|------|-----------|
+| [`fixtures.toml`](fixtures.toml) | Global constants (seed, master fingerprint, network). |
+| [`get_extended_pubkey.toml`](get_extended_pubkey.toml) | `get_extended_pubkey`. |
```
Comment (iartemov-ledger) [inline test_vectors/README.md]: > `get_extended_pubkey`.
The point at the end looks weird when rendered: https://github.com/LedgerHQ/app-bitcoin-new/blob/test_assets/test_vectors/README.md.

File: test_vectors/get_extended_pubkey.toml [hunk 2]
```diff
@@ -0,0 +1,211 @@
+# Test vectors for get_extended_pubkey.
+#
+# Source: tests/test_get_extended_pubkey.py
+#
+# Schema (per [[case]]):
```
Comment (iartemov-ledger) [inline test_vectors/get_extended_pubkey.toml:5]: There are apparently approaches to validate a .toml through Standard JSON Schema validator. There is also a tool callled "Taplo" for this. 
Just for INFO.

**Comment by iartemov-ledger** [comment]
## Test coverage impact

Column format in the source report: **total lines, covered lines, [two uncovered sub-counts], coverage %** (e.g. `register_wallet.c 80 63 7 10 → 63/80 = 78.75%`).

### 1. Coverage changed in existing files

Files present in both runs whose coverage % moved (covered/total shown):

| File | Before | After | Δ |
|------|--------|-------|---|
| `src/handler/lib/policy.c` | 42.57% (361/848) | 48.94% (415/848) | **+6.37** |
| `src/common/wallet.c` | 77.68% (1246/1604) | 78.18% (1254/1604) | +0.50 |
| `src/crypto.c` | 68.85% (168/244) | 69.26% (169/244) | +0.41 |

> All other shared files were unchanged in %: `musig.c` (37.41), `parser_ext.c` (46.34), `segwit_addr.c` (63.20), `script.c` (79.51), `psbt_parse.c` (60.98), `sign_psbt_cache.c` (41.67), `crypto.h` (58.82), `xpub.c` (53.33), and the many 100% merkle/stream/buffer files.
> `src/boilerplate/dispatcher.h` stayed 100% but its counted LOC grew 3→16, since the three new handlers instantiate more of it.

### 2. Files newly reported on (after only)

Files that appear in the *after* report but not *before* — the new handler sources now exercised, plus the new test/harness files:

| File | Coverage | Covered/Total |
|------|----------|---------------|
| **Production handlers (newly exercised)** | | |
| `src/handler/register_wallet.c` | 78.75% | 63/80 |
| `src/handler/get_extended_pubkey.c` | 74.65% | 53/71 |
| `src/handler/sign_message.c` | 71.23% | 52/73 |
| **New test / harness code** | | |
| `unit-tests/test_get_extended_pubkey.c` | 70.00% | 105/150 |
| `unit-tests/test_sign_message.c` | 68.78% | 152/221 |
| `unit-tests/test_register_wallet.c` | 68.18% | 135/198 |
| `unit-tests/libs/toml_helpers.h` | 40.00% | 14/35 |

### Takeaways

- The headline production wins are the three handlers going from **0 → ~71–79%**. The uncovered remainder (~21–29%) is the untested `SW_DENY` UI-reject paths, defensive `SW_BAD_STATE` branches, and a few framing/edge branches (`display > 1`, `WRONG_DATA_LENGTH`).
- `policy.c` (**+6.37**) is the biggest *indirect* gain — register_wallet's sanity/classification checks drive much of it; adding the missing extended-error-code (`EC_*`) assertions would likely lift it further.
- `toml_helpers.h` at 40% is expected — it's a header-only menu of helpers, and unused-in-this-binary inline functions still count against the file.

File: unit-tests/CMakeLists.txt [hunk 3]
```diff
@@ -332,15 +332,72 @@ if(SPECULOS AND SPECULOS_SRC)
   add_test(test_wallet test_wallet)
```
Comment (iartemov-ledger) [inline unit-tests/CMakeLists.txt:333]: Siggestion from Claude  to add Address + UndefinedBehavior Sanitizer here as low hanging fruits:

```
diff --git a/unit-tests/CMakeLists.txt b/unit-tests/CMakeLists.txt
index 4316dd86..3077dbd0 100644
--- a/unit-tests/CMakeLists.txt
+++ b/unit-tests/CMakeLists.txt
@@ -27,6 +27,25 @@ set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -Wall -pedantic -g -O0 --coverag
 set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} --coverage")
 set(CMAKE_EXE_LINKER_FLAGS    "${CMAKE_EXE_LINKER_FLAGS} --coverage")
 
+# Opt-in AddressSanitizer + UndefinedBehaviorSanitizer. These tests are full
+# of the patterns ASan/UBSan catch best (a hand-rolled base64 decoder,
+# fixed-buffer memcpy, Merkle-chunk pointer math, VLAs sized from a vectors
+# file), so this is cheap insurance. Off by default to keep the plain build
+# fast; enable with -DSANITIZE=ON.
+#
+# The flags go on the global compile/link flags (not per-target) so the SHARED
+# helper libs are instrumented too — otherwise bugs inside them go unseen and
+# partial instrumentation can break linking. -fno-sanitize-recover=all makes
+# UBSan abort (non-zero exit) instead of printing and continuing, so CI fails
+# on the first finding. tomlc17 keeps its own -w but is still instrumented.
+option(SANITIZE "Build with AddressSanitizer + UndefinedBehaviorSanitizer" OFF)
+if(SANITIZE)
+  set(SANITIZE_FLAGS "-fsanitize=address,undefined -fno-sanitize-recover=all -fno-omit-frame-pointer")
+  set(CMAKE_C_FLAGS_DEBUG       "${CMAKE_C_FLAGS_DEBUG} ${SANITIZE_FLAGS}")
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${SANITIZE_FLAGS}")
+  set(CMAKE_EXE_LINKER_FLAGS    "${CMAKE_EXE_LINKER_FLAGS} ${SANITIZE_FLAGS}")
+endif()
+
 if(CMAKE_SOURCE_DIR STREQUAL CMAKE_BINARY_DIR)
   message(FATAL_ERROR "In-source builds not allowed. Please make a new directory (called a build directory) and run CMake from there. You may need to remove CMakeCache.txt.")
 endif()

```

To be built with `-DSANITIZE=ON`.

I tried and it gava me several errors (see in attachment) , to be considered at some point: 
[out.txt](https://github.com/user-attachments/files/28999399/out.txt)

File: unit-tests/test_get_extended_pubkey.c [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+ * "(Master key)" for the empty path. */
+static const char *expected_path_str(const char *path) {
+    if (strcmp(path, "m") == 0) {
+        return "(Master key)";
+    }
+    return path + 2;  // skip "m/"
+}
+
+/* Map an error name from the vectors file to its status word. */
+static uint16_t error_name_to_sw(const char *name) {
```
Comment (iartemov-ledger) [inline unit-tests/test_get_extended_pubkey.c:204]: No check for extended error `EC_REGISTER_WALLET_*`.

**Review by iartemov-ledger — DISMISSED** [review]
Cool work!

Approved with minor comments.

File: test_vectors/README.md (hunk 1 above)
Comment (bigspider) [reply test_vectors/README.md]: Indeed, fixed in 33ef4070fb0c19bd0130286b9077d308d7ba30c1.

**Review by iartemov-ledger — APPROVED** [review]
